src/main/java/ru/yandex/practicum/filmorate/service/DBFilmService.java

```diff
@@ -93,13 +105,48 @@ public List<Film> getFilms() {
         List<Film> films = new ArrayList<>();
         for (Film film : filmDao.getFilms()) {
             film.setGenres(genreDao.getFilmGenres(film.getId()));
+            directorDao.addDirectorsListToFilm(film);
             films.add(film);
         }
         return films;
     }
 
+    public List<Film> getDirectorsFilmsByRate(Integer directorsId) {
+        directorDao.checkDirectorExist(directorsId);
+        List<Film> directorFilms = new ArrayList<>();
+        for (Film film : filmDao.findDirectorsFilmsSortedByRate(directorsId)) {
+            film.setGenres(genreDao.getFilmGenres(film.getId()));
+            directorFilms.add(film);
+        }
+        return directorFilms;
+    }
+
+    public List<Film> getDirectorsFilmsByYears(Integer directorsId) {
+        directorDao.checkDirectorExist(directorsId);
+        List<Film> directorFilms = new ArrayList<>();
+        for (Film film : filmDao.findDirectorsFilmsSortedByYears(directorsId)) {
+            film.setGenres(genreDao.getFilmGenres(film.getId()));
+            directorFilms.add(film);
+        }
+        return directorFilms;
+    }
+
     public List<Film> getTopFilms(Integer count) {
         log.info("Get {} popular films", count);
         return filmDao.getTopFilms(count);
     }
+
+    public void addDirectorsToFilm(Film film){
```

Comment:
Checkstyle. '{' не содержит пробелов.
```suggestion
    public void addDirectorsToFilm(Film film) {
```